pkg/vm/stackitem/item.go

```diff
@@ -1139,12 +1171,12 @@ func (i *Buffer) Len() int {
 // DeepCopy returns a new deep copy of the provided item.
 // Values of Interop items are not deeply copied.
 // It does preserve duplicates only for non-primitive types.
-func DeepCopy(item Item) Item {
+func DeepCopy(item Item, asImmutable bool) Item {
```

Comment:
`asImmutable` doesn't make a copy (except for `Buffer`), so we better have `DeepCopy` that does a deep copy and `MakeImmutable` that makes `item` immutable. Just to make the external interface more clear.

Reply:
> `asImmutable` doesn't make a copy (except for Buffer)

But why? Insede `DeepCopy` we firstly perform item copying, and after that mark this new copy as immutable. We perform this for each stackitem.

Also, we *always* need to mark as immutable arrays, structs and maps while deepcopying irrespective to `asImmutable`.

Reply:
OK, my original impression of the C# implementation was wrong.